components/provisioner/internal/model/gardener_config.go

```diff
@@ -590,7 +591,42 @@ func (c AWSGardenerConfig) ValidateShootConfigChange(shoot *gardener_types.Shoot
 }
 
 func (c AWSGardenerConfig) EditShootConfig(gardenerConfig GardenerConfig, shoot *gardener_types.Shoot) apperrors.AppError {
-	return updateShootConfig(gardenerConfig, shoot)
+	err := updateShootConfig(gardenerConfig, shoot)
+	if err != nil {
+		return err
+	}
+
+	if c.input.EnableIMDSv2 != nil && *c.input.EnableIMDSv2 {
+		var (
+			workerConfig            *aws.WorkerConfig
+			httpPutResponseHopLimit int64 = 2
+		)
+		if shoot.Spec.Provider.Workers[0].ProviderConfig == nil {
+			workerConfig = NewAWSWorkerConfig(2)
```

Comment:
Instead of passing the magic number we can use, defined couple lines above, variable `httpPutResponseHopLimit`. That will improve code transparency.

Reply:
Introduced global variable for AWS IMDSv2 HTTPPutResponseHopLimit, and it's used everywhere instead of the "2" magic number.